project_timesheet_holidays_contract/models/hr_employee.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import datetime
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    def _get_datetime_date(self, date):
+        return date if not isinstance(date, datetime) else date.date()
+
+    def _get_calendar(self, date_from=None):
+        res = super()._get_calendar(date_from)
+        if not date_from:
+            return res
+        date_from = self._get_datetime_date(date_from)
+        date_to = self.env.context.get("date_to", fields.Date.today())
+        date_to = self._get_datetime_date(date_to)
+        contracts = (
+            self.sudo()
+            .with_context(active_test=True)
+            .env["hr.contract"]
+            .search(
+                [
+                    ("employee_id", "=", self.id),
+                    ("date_start", "<=", date_from),
+                    "|",
+                    ("date_end", ">=", date_to),
+                    ("date_end", "=", False),
+                ]
+            )
+        )
+        if contracts:
+            # Filter out cancelled contracts
+            contracts = contracts.filtered(lambda c: c.state != "cancel")
+            if len(contracts) == 1:
+                return contracts[0].resource_calendar_id
+            # Multiple non-cancelled contracts for this date period
+            if contracts:
+                raise ValidationError(_("This period overlaps multiple contracts!"))
```

Comment:
In the previous version of the code:
- if only 1 contract was retrieved by `search()`, it was returned regardless of its state;
- if there were more than 1, then the contracts would've been filtered by `state != "cancel"`, and if there were only 1 non-cancelled contracts, it would've been returned, else an error would be raised

Now:
- contracts are filtered by state beforehand, and if there's only 1 non-cancelled contract, it is returned
- if there are multiple non-cancelled contracts, an error is raised
- if all retrieved contracts are cancelled, nothing happens and `super()` is called

Is this change in the code behavior expected?

Reply:
@SilvioC2C I don't have the requirements nor the exact behavior. The original request was to deliver this module as an OCA one instead of a custom.